### ISSUE TITLE
Zamia ASR pre-trained model.

### ISSUE THREAD

**Issue by Cloud299** [pykaldi/pykaldi#79]
I'm trying to use of the pre-trained ASR [speech to text models](http://goofy.zamia.org/zamia-speech/asr-models/) on a test wav file in Python. What's the easiest way to do this ? I just finished installing pykaldi. Could someone provide me a snippet of code on how to do this ? Say I want to use the kaldi-generic-en-tdnn_f model.

**Comment by dogancan** [comment]
Try following [this example](https://github.com/pykaldi/pykaldi/tree/master/examples/setups/aspire). It should be fairly straightforward to replace the models with your own.

**Comment by dogancan** [comment]
I just added an [example setup for Zamia models](https://github.com/pykaldi/pykaldi/tree/master/examples/setups/zamia). Let me know if you have any trouble running it.

**Comment by Cloud299** [comment]
Thank you very much. It's working well.

**Comment by panagiotismavridis** [comment]
Thanks for also including me into this!

I suggest they use mainly the docker image. Works more easily.

Panagiotis

**Comment by Cloud299** [comment]
> I suggest they use mainly the docker image

Could you elaborate on this ? 

Also, I was wondering if there'd be any issue with multiprocessing (decoding multiple files in parallel?) and if pykaldi can run on Mac as well ? I'd really appreciate it and I think a lot of people would benefit from being able to run this on Mac.

**Comment by dogancan** [comment]
Docker image is useful if you have trouble installing from source or if you want to quickly spawn a container with a pykaldi installation.

There shouldn’t be any issues with multiprocessing. If there is let me know. 

Currently pykaldi is Linux only. It shouldn’t be too hard to get it working on Mac but it is not a priority for us at the moment. Pull requests are welcome :)

**Comment by dogancan** [comment]
@Cloud299 Just added support for macOS. I would appreciate if you could try installing on Mac and report what happens.

**Comment by Cloud299** [comment]
Thank you for including support for macOS. I'm running into the following error when doing /install_clif.sh 

 **clang: error: invalid deployment target for -stdlib=libc++ (requires OS X 10.7 or later)
    error: command 'gcc' failed with exit status 1**

I'm running macOS 10.14.3

**Comment by dogancan** [comment]
Is this a clean installation? I need to see more context to understand at what stage of the build you are receiving this error.

**Comment by dogancan** [comment]
Can you try running the following before install_clif.sh

```bash
export MACOSX_DEPLOYMENT_TARGET=10.14
```

**Comment by dogancan** [comment]
Great! No idea why cmake did not automatically pick up the deployment target based on your host system though.

**Comment by Cloud299** [comment]
I just updated the folder "zamia" to include the new model from 2019 on my Mac. However I'm running into a "Cannot open config file: conf/ivector_extractor.conf". I suspect this has to do with the command "ln -sr exp/nnet3_chain/ivectors_test_hires/conf/ivector_extractor.conf conf/." and "ln -sr ../aspire/data/test data/." from the update model.sh.

I don't think the flag -sr applies for Mac. How should I fix this ?

**Comment by Cloud299** [comment]
Never mind. Fixed this by simply copying exp/nnet3_chain/ivectors_test_hires/conf/ivector_extractor.conf to conf/

**Comment by dogancan** [comment]
The "-sr" flag should work on Mac. You probably downloaded the new models into a dirty directory containing old models and symbolic linking somehow failed. The model download scripts are bare bones, they don't check if things are already in place before trying to override them.

**Comment by Cloud299** [comment]
I see thank you for the feedback. 

I have another question. I wrote a script "test.py" that imports a custom function I wrote in decode.py in /examples/setups/aspire/. This function works perfectly fine in /aspire folder. However, when I execute this script from outside the /aspire folder, I get the following error: 

<img width="983" alt="Screen Shot 2019-03-08 at 2 31 50 PM" src="https://user-images.githubusercontent.com/41118094/54054620-4f2e6180-41b0-11e9-90d3-8779e2e20ff6.png">

I made sure that all the paths in decode.py are full and global. I think this error might have to do with the file path.sh but I'm not sure how to fix this yet. 

Thank you for all your answers!

**Comment by dogancan** [comment]
Seems like you didn't update the paths in [conf/ivector_extractor.conf](https://github.com/pykaldi/pykaldi/blob/master/examples/setups/aspire/conf/ivector_extractor.conf).

**Comment by Cloud299** [comment]
Thank you very much. This fixed the problem. I really like how you always got to the crux of the problem immediately.

**Comment by dogancan** [comment]
You're welcome.

**Comment by Cloud299** [comment]
Just for curiosity, in decode.py, is there an option to remove all the verbose when running the process ?

**Comment by dogancan** [comment]
Almost all of the verbosity is coming from kaldi so there is no way of suppressing all of the verbosity without changing kaldi code. All of the log messages are written to stderr so you can separate log messages from script output by redirecting standard error to a log file, e.g.

```
./decode.py >log.out 2> log.err
```
